### ISSUE TITLE
[a11y]: Date Picker Calendar Tab Index

### ISSUE THREAD

**Issue by ibm-sint** [carbon-design-system/carbon#22579]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11.110.0

### React version

v19

### Automated testing tool and ruleset

Keyboard

### Description

When tabbing on the Date Picker with Calendar (https://react.carbondesignsystem.com/?path=/story/components-datepicker--single-with-calendar), it is not possible to tab to the buttons to change the month forward/backwards meaning that if I want to use the calendar to select a date, I will need to scroll through all the days and cannot scroll through the months as a visual/mouse user is able to do. Therefore, the calendar component's month forward/back buttons are not accessible to a keyboard.

Even if a visually impaired keyboard user types the date in to the input with a keyboard, because the calendar is automatically opened (which they are unaware of) on pressing tab to select the next form element, instead of selecting the next form element the calendar will be selected. There is no affordance to a visually impaired user why they are now in a calendar. There should be a message or label to explain to this user that the calendar is for the input element prior to them pressing tab, or the calendar should only be shown e.g. on click of the icon on the right-hand-side of the input element (with affordance to the user with Assistive Technology that the button will show a selectable calendar for the element - example of that design here: https://design-patterns.service.justice.gov.uk/components/date-picker)

https://github.com/user-attachments/assets/3bd48454-5abf-428f-b921-a9bae4c26276

### WCAG 2.1 Violation

1.3.2 Meaningful Sequence, 2.1.1 Keyboard, 2.4.3 Focus Order

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datepicker--single-with-calendar

### Steps to reproduce

In a DatePicker component with Calendar, press the tab button to try and tab to the next/prev month buttons or to get to the next element in a form (e.g. next input or the submit button)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by ibm-sint** [comment]
@heloiselui @tay1orjones I've raised this one on behalf of my client. I'm available on Slack as well if any further information is needed and can hop on a call even if what I've put here isn't clear.

**Comment by maradwan26** [comment]
Hello @ibm-sint! We're going to run this by our accessibility team to get some further guidance and input on behaviours and requirements and get back to you.

**Comment by maradwan26** [comment]
Hi @ibm-sint, got some updates:

Our DatePicker component uses flatpickr internally, which has documented issues with its keyboard nav and accessibility such as the issues you've mentioned: https://github.com/flatpickr/flatpickr/issues/1694, https://github.com/flatpickr/flatpickr/issues/1695. One of which is a bug with the keyboard shortcut for changing months/years. Typically datepicker calenders use `PgUp` and `PgDn` keys to change months and add `Shift` to change years, but flatpickr uses CTRL + ↑/→, ~~which on MacOS conflicts with the Desktop switch shortcut. You can test this out by checking out the component on the Windows machine and seeing the CTRL shortcuts will work there.~~

 ⬆️  (correction): It is CTRL + OPTION + ↑/→ on MacOS. So, on both Windows and MacOS you can use a keyboard shortcut to navigate through the months/years. 

The Carbon Team is planning to move away from using flatpickr. We have an experimental DatePicker component in [Carbon Labs](https://labs.carbondesignsystem.com/v12/?path=/docs/web-components_v12-date-picker--docs) that doesn't rely on flatpickr. We plan to graduate this as a Preview component in the Core library (https://github.com/carbon-design-system/carbon/issues/22427), where we will then work on stabilizing it by fixing/addressing issues (like this one for example) and receiving any feedback.

Regarding the accessibility guidance received:
- Previous/Next buttons, and Year stepper should be tabbable
- DatePicker should announce year/month changes
- DatePicker should support `PgUp`, `PgDn` shortcuts
- DatePicker should announce opening (probably just through via aria-expanded)

**Comment by ibm-sint** [comment]
> * DatePicker should announce opening (probably just through via aria-expanded)

Will this be automatic, or only if the user chooses to e.g. by clicking on the calendar button? Wouldn't it be more accessible to allow the user to trigger the calendar pop-up, rather than it being automatic without the user wanting to just by tabbing into the field?

**Comment by chrisgrimley** [comment]
Hi @ibm-sint. I agree that having two separate actions for date input and calendar input might provide a cleaner, more understandable experience, but I also don't see any violation or major accessibility issue. Even though the calendar automatically opens, the focus remains on the input field. Also AT doesn't announce the calendar. It therefore doesn't interrupt the user flow or change user context.  I checked this with colleagues in the A11y team and they agreed that the automatic expansion of the calendar is ok.
